docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_traffic_to_origin_unencrypted_remediation.mdx

```diff
@@ -1,6 +1,228 @@
+
+4. **Require HTTPS:**
+   - In the behavior settings, find the "Viewer Protocol Policy" option.
+   - Select "Redirect HTTP to HTTPS" or "HTTPS Only" from the dropdown menu.
+   - Click the "Yes, Edit" button to save your changes.
+
+By following these steps, you ensure that your CloudFront distribution will use HTTPS for secure delivery of web content.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that your CloudFront distributions use HTTPS for secure delivery of web content using AWS CLI, follow these steps:
+
+1. **List CloudFront Distributions:**
+   First, identify the CloudFront distribution that you want to configure to use HTTPS.
+
+   ```sh
+   aws cloudfront list-distributions
+   ```
+
+2. **Get Distribution Configuration:**
+   Retrieve the current configuration of the specific CloudFront distribution. Replace `DISTRIBUTION_ID` with your actual distribution ID.
+
+   ```sh
+   aws cloudfront get-distribution-config --id DISTRIBUTION_ID
+   ```
+
+3. **Update Distribution Configuration:**
+   Modify the distribution configuration to enforce HTTPS. You need to set the `ViewerProtocolPolicy` to `redirect-to-https` or `https-only` for each cache behavior. First, save the current configuration to a file (e.g., `config.json`), then edit the file to update the `ViewerProtocolPolicy`.
+
+   ```sh
+   aws cloudfront get-distribution-config --id DISTRIBUTION_ID > config.json
+   ```
+
+   Edit `config.json` to set `ViewerProtocolPolicy` to `redirect-to-https` or `https-only` for each cache behavior.
+
+4. **Apply Updated Configuration:**
+   Apply the updated configuration back to the CloudFront distribution. Ensure you include the `If-Match` header with the ETag value from the previous configuration to avoid conflicts.
+
+   ```sh
+   aws cloudfront update-distribution --id DISTRIBUTION_ID --distribution-config file://config.json --if-match E_TAG
+   ```
+
+Replace `DISTRIBUTION_ID` with your CloudFront distribution ID and `E_TAG` with the ETag value obtained from the `get-distribution-config` command.
+
+By following these steps, you can ensure that your CloudFront distributions are configured to use HTTPS for secure delivery of web content.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that CloudFront distributions use HTTPS for secure delivery of web content, you can use the AWS SDK for Python (Boto3) to configure your CloudFront distributions accordingly. Below are the steps to prevent this misconfiguration using a Python script:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session and create a CloudFront client.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_AWS_ACCESS_KEY_ID',
+       aws_secret_access_key='YOUR_AWS_SECRET_ACCESS_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>